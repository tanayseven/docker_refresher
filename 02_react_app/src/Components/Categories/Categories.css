.Categories {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 30px;
}

.Categories li {
    font-size: 30px;
    margin-bottom: 30px;
    max-width: 300px;
    width: 100%
}

.Categories li a {
    padding: 10px 20px;
    display: block;
    text-align: center;
}
