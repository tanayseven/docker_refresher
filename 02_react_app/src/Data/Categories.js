
const Categories = [
    {
        text: "Addition",
        uri: "/add"
    },
    {
        text: "Subtraction",
        uri: "/sub"
    },
    {
        text: "Multiply",
        uri: "/multiply"
    },
    {
        text: "Comparison",
        uri: "/comparison"
    },
    // {
    //     text: "Division",
    //     uri: "/division"
    // }
]

export default Categories