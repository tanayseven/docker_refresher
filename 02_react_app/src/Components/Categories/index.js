import React, { Component } from 'react';
import Header from './../Header'
import { Link } from "react-router-dom";
import CategoriesData from './../../Data/Categories'
import './Categories.css'

class Categories extends Component {
    render() {
        return (
            <div className="Categories__Wrapper">
                <Header />
                <ul className="Categories">
                    {
                        CategoriesData.map((category, i) => {
                            return (
                                <li className="Wood" key={`${category.text}-${i}`}>
                                    <Link to={category.uri}>{category.text}</Link>
                                </li>
                            )
                        })
                    }
                </ul>
            </div>
        );
    }
}

export default Categories;