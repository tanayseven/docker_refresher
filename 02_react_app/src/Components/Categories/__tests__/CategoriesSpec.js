import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter} from 'react-router-dom'
// import { shallow, render } from 'enzyme';
import Categories from './../index';

describe('Categories', () => {
    it('renders correctly', () => {
        let tree = renderer.create(
            <StaticRouter location="someLocation" context={{}}>
                <Categories  />
            </StaticRouter>
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });
});