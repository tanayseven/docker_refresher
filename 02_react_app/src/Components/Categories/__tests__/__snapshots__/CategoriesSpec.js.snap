// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories renders correctly 1`] = `
<div
  className="Categories__Wrapper"
>
  <header
    className="Header"
  >
    <div
      className="App__Logo"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Speedy Math
      </a>
    </div>
  </header>
  <ul
    className="Categories"
  >
    <li
      className="Wood"
    >
      <a
        href="/add"
        onClick={[Function]}
      >
        Addition
      </a>
    </li>
    <li
      className="Wood"
    >
      <a
        href="/sub"
        onClick={[Function]}
      >
        Subtraction
      </a>
    </li>
    <li
      className="Wood"
    >
      <a
        href="/multiply"
        onClick={[Function]}
      >
        Multiply
      </a>
    </li>
    <li
      className="Wood"
    >
      <a
        href="/comparison"
        onClick={[Function]}
      >
        Comparison
      </a>
    </li>
  </ul>
</div>
`;
